### PR TITLE
Allow custom replica announce IP/port and Sentinel announce IP

### PR DESCRIPTION
**Pull request by pierre-claranet**
#### Pull Request (PR) description
Add parameters `replica-announce-ip` and `replica-announce-port` for Redis, and `sentinel announce-ip` for Redis-Sentinel

These parameters are usefull with NAT, and when TLS is enabled.

### CODE REVIEW

File: templates/redis.conf.epp [hunk 1]
```diff
@@ -432,6 +434,37 @@ slave-serve-stale-data <%= bool2str($slave_serve_stale_data, 'yes', 'no') %>
...
+# report to its master a specific set of IP and port, so that both INFO
+# and ROLE will report those values.
+#
+# There is no need to use both the options if you need to override just
+# the port or the IP address.
+#
+# replica-announce-ip 5.5.5.5
+# replica-announce-port 1234
+<% if $repl_announce_ip { -%>replica-announce-ip <%= $repl_announce_ip %><% } -%>
+<% if $repl_announce_port { -%>replica-announce-port <%= $repl_announce_port %><% } %>
```
Comment (smortex) [inline templates/redis.conf.epp:466]: Maybe for consistency:

```suggestion
<% if $repl_announce_port { -%>replica-announce-port <%= $repl_announce_port %><% } -%>
```

**Comment by bastelfreak** [comment]
thanks for the work. Can you add some unit tests to verify the templates render correctly?

File: templates/redis.conf.epp (hunk 1 above)
Comment (pierre-claranet) [reply templates/redis.conf.epp]: Hello Smortex, you're right, it seems better. I fixed it.

**Comment by bastelfreak** [comment]
Thanks for the work!

**Comment by pierre-claranet** [comment]
Thank you !

**Pull request merged**